src/tools/rbd/action/Bench.cc

```diff
@@ -246,15 +269,24 @@ int do_bench(librbd::Image& image, io_type_t io_type,
 
   vector<uint64_t> thread_offset;
   uint64_t i;
-  uint64_t start_pos;
+  uint64_t start_pos = 0;
 
   uint64_t unit_len = size/io_size/io_threads;
   // disturb all thread's offset
   for (i = 0; i < io_threads; i++) {
-    if (random) {
+    switch (io_pattern) {
+    case IO_PATTERN_RAND:
       start_pos = (rand() % (size / io_size)) * io_size;
-    } else {
+      break;
+    case IO_PATTERN_SEQ:
       start_pos = unit_len * i * io_size;
+      break;
+    case IO_PATTERN_FULL_SEQ:
+      start_pos = i * io_size;
+      unit_len = io_size * io_threads;
```

Comment:
nit: The "physical" meaning of `unit_len` was already confusing to me and now it becomes even more. Initially it is defined as `size/io_size/io_threads`, so the unit for this "len" var is [ bytes ] / [ bytes ] / [ num ] = 1 / [num]. And now the unit becomes [bytes] * [num]. Something like assigning a speed value to a "time" variable.

Probably just remove usage of this `unit_len` variable at all and do direct calculations when needed?

Reply:
I created a new ```seq_chunk_length``` variable (in byte units now) and it's only used for seq IO since I didn't want to recompute it in three places.